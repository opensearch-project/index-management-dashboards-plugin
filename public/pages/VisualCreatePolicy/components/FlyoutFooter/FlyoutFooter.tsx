/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { EuiFlexGroup, EuiFlexItem, EuiButtonEmpty, EuiButton } from "@elastic/eui";

interface FlyoutFooterProps {
  edit: boolean;
  action: string;
  disabledAction?: boolean;
  onClickCancel: () => void;
  onClickAction: () => void;
  save?: boolean;
  restore?: boolean;
}

const FlyoutFooter = ({ edit, action, disabledAction = false, onClickCancel, onClickAction, save, restore }: FlyoutFooterProps) => (
  <EuiFlexGroup justifyContent="flexEnd">
    <EuiFlexItem grow={false}>
      <EuiButtonEmpty onClick={onClickCancel} flush="left" data-test-subj="flyout-footer-cancel-button">
        Cancel
      </EuiButtonEmpty>
    </EuiFlexItem>
    <EuiFlexItem grow={false}>
      <EuiButton disabled={disabledAction} onClick={onClickAction} fill data-test-subj="flyout-footer-action-button">
        {restore ? "Restore snapshot" : !save ? `${edit ? "Edit" : "Add"} ${action}` : save ? "Save" : "Create"}
      </EuiButton>
    </EuiFlexItem>
  </EuiFlexGroup>
);

export default FlyoutFooter;
