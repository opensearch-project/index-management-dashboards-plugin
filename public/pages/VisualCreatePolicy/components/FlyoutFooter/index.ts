/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import FlyoutFooter from "./FlyoutFooter";

export default FlyoutFooter;
