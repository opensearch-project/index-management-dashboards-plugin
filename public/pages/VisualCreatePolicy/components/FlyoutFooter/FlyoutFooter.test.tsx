/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { fireEvent } from "@testing-library/dom";
import FlyoutFooter from "./FlyoutFooter";

describe("<FlyoutFooter /> spec", () => {
  it("renders the component", () => {
    const { container } = render(<FlyoutFooter edit={true} action="action" onClickAction={() => {}} onClickCancel={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onClickAction  with clicking action button", async () => {
    const onClickAction = jest.fn();
    const { getByTestId } = render(<FlyoutFooter edit={true} action="action" onClickAction={onClickAction} onClickCancel={() => {}} />);

    fireEvent.click(getByTestId("flyout-footer-action-button"));
    expect(onClickAction).toHaveBeenCalledTimes(1);
  });

  it("calls onClickCancel  with clicking cancel button", async () => {
    const onClickCancel = jest.fn();
    const { getByTestId } = render(<FlyoutFooter edit={true} action="action" onClickAction={() => {}} onClickCancel={onClickCancel} />);

    fireEvent.click(getByTestId("flyout-footer-cancel-button"));
    expect(onClickCancel).toHaveBeenCalledTimes(1);
  });
});
