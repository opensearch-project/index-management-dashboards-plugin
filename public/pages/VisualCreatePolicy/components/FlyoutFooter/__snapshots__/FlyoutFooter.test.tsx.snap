// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlyoutFooter /> spec renders the component 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <button
      class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small euiButtonEmpty--flushLeft"
      data-test-subj="flyout-footer-cancel-button"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonEmpty__content"
      >
        <span
          class="euiButtonEmpty__text"
        >
          Cancel
        </span>
      </span>
    </button>
  </div>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <button
      class="euiButton euiButton--primary euiButton--small euiButton--fill"
      data-test-subj="flyout-footer-action-button"
      type="button"
    >
      <span
        class="euiButtonContent euiButton__content"
      >
        <span
          class="euiButton__text"
        >
          Edit action
        </span>
      </span>
    </button>
  </div>
</div>
`;
